app/api/helpers/scheduled_jobs.py

```diff
@@ -46,6 +47,16 @@ def send_after_event_mail():
                     send_notif_after_event(organizer.user.email, event.name)
 
 
+def change_session_state_on_event_completion():
+    from app import current_app as app
+    with app.app_context():
+        sessions_to_be_changed = Session.query.join(Event).filter(Session.state == 'pending')\
+                                 .filter(Event.ends_at < datetime.datetime.now())
+        for session in sessions_to_be_changed:
+            session.state = 'rejected'
+            save_to_db(session, 'Changed {} session state to rejected'.format(session.title))
```

Comment:
No need for looping, can be done in a single query

Reply:
@iamareebjamal I tried using a loop directly to test the updates with 

```python
db.session.query(Session).join(Event).filter(Session.state=='confirmed').update({"video_url":'a'})
db.session.commit()
```

and it threw : 

```
sqlalchemy.exc.InvalidRequestError: Can't call Query.update() or Query.delete() when join(), outerjoin(), select_from(), or from_self() has been called

```

How do i tackle this?

Reply:
you will not be able to use an update alongside a successive join. Why do you need to use `join` here? You can directly use the related `Session`'s event's fields.

Reply:
https://stackoverflow.com/questions/33153823/update-joined-table-via-sqlalchemy-orm-using-session-query

Reply:
@uds5501 use `db.session.query(Session).filter_by(event_id=your_event_id, state=='confirmed').update(...)`

Reply:
@mrsaicharan1 this query won't check if the event has passed